File: alarm-dashboard-backend/src/main/java/com/alarmdashboard/alarm_dashboard_backend/model/AlarmEvent.java
package com.alarmdashboard.alarm_dashboard_backend.model;

import java.time.Instant;

public record AlarmEvent(
        String id,
        String level,     // CRITICAL | WARN | INFO
        String type,      // e.g. FAN_FAILURE
        String location,  // e.g. Tunnel-3
        String message,
        Instant timestamp
) {}

File: alarm-dashboard-frontend/src/app/core/realtime/alarm-event.ts
export interface AlarmEvent {
  id: string;
  level: 'CRITICAL' | 'WARN' | 'INFO';
  type: string;
  location: string;
  message: string;

  timestamp: string;   // ISO-8601 (payload)
  arrivedAt?: string;  // ISO-8601 (UI pencereleri bunu kullanır)
  createdAt?: string;  // ISO-8601

  // UI için türetilmiş (path'ten çıkarılır)
  system?: string;     // EmergencyCall / LVMV / Lighting ...
  device?: string;     // Call / CircuitBreaker / LightLevel ...
  point?: string;      // T1RC3 / AKMADPQ1 ...
}